jonason/Tokenizer.hpp

```diff
@@ -0,0 +1,59 @@
+//
+//  Tokenizer.hpp
+//  jonason
+//
+//  Created by Gabriel Lumbi on 2023-03-02.
+//
+
+#ifndef Tokenizer_hpp
+#define Tokenizer_hpp
+
+#include <vector>
+#include <istream>
+#include <string>
+
+namespace jonason {
+
+struct Token {
+
+public:
+    enum {
+        LITERAL = '*',
+
+        OBJECT_OPEN = '{',
+        OBJECT_CLOSE = '}',
+
+        ARRAY_OPEN = '[',
+        ARRAY_CLOSE = ']',
+
+        DOUBLE_QUOTE = '"',
+        COLUMN = ':',
+        COMMA = ','
+    }
+    tag;
+
+    char* value = nullptr;
+
+    using Type = decltype(Token::LITERAL);
```

Comment:
Neat trick with anonymous enums, I hadn't seen it. My default would be to use an `enum class`.